### PR TITLE
List view

### PR DESCRIPTION
**Pull request by nachtm**
This PR adds the list view, which looks like this on desktop:
image
... and this on mobile:
image

The ListCard expects the following fields to be present in the result we get from firebase:
 - location: {_latitude: , _longitude}
 - species: string
 - time_seen: timestamp (defaults to time_submitted if not provided) (should it default?)
 - image_url: url string (defaults to our default image, currently a bear stolen from wikipedia)

I also added a REACT_APP_GET_REPORTS field to my .env file pointing to a local instance of firebase functions. I'm not sure if that's the best way to go about it or if I don't even need to worry about using .env for that kind of thing. 

Things that need to be done before this is totally complete:
 - [ ] get a better (and local) default image
 - [ ] convert locations into neighborhoods
 - [ ] scrolling with the footer div
 - [ ] footer div background color?
 - [ ] actually deploy getReports to firebase (have to resolve a couple of linting issues)

### CODE REVIEW

File: src/components/ListCard.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+const ListCard = (props) => {
+  const { classes, data } = props;
+  return <Card className={classes.card}>
+    <CardContent className={classes.allContent}>
+      <CardMedia className={classes.picture}
+        image={data.image_url ? data.image_url : DEFAULT_IMAGE_URL}
+      />
+      <CardContent className={classes.info}>
+        <Typography variant={'h3'}>{data.species.toUpperCase()}</Typography>
+        <Typography variant={'subtitle1'}>{timeToString(data.time_seen ? data.time_seen : data.time_submitted)}</Typography>
```
Comment (sayas01) [inline src/components/ListCard.js]: Isn't `time_seen` a required field?

Comment (nachtm) [reply src/components/ListCard.js]: It should be -- it wasn't on a lot of the fields currently in the db, so I decided to handle it cleanly. But I agree that ultimately every report should have a `time_seen` field. Should I assume `time_seen` exists?

Comment (sayas01) [reply src/components/ListCard.js]: Yes, this makes sense as most of the current db entries don't have this field but I would prefer to make the `time_seen` field a required field when the form is submitted. @savannahostrowski what do you think?

Comment (nachtm) [reply src/components/ListCard.js]: I went through and added the time_seen field to the db entries so it still looks good, and updated this accordingly :)

File: src/components/ListView.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  container: {
+    backgroundColor: 'grey',
+    display: 'flex',
+    alignItems: 'center',
+    flexDirection: 'column'
+  }
+});
+
+class ListView extends React.Component {
+  constructor(props) {
```
Comment (savannahostrowski) [inline src/components/ListView.js]: We can pull this out of the constructor so it's just `state={};` at the top of the class.

File: src/components/ListView.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        this.setState({...this.state, reports})
+      })
+      .catch(error => error);
+  }
+
+  render() {
+    return (
+    <div className={this.props.classes.container}>
+    {
+      this.state.reports
```
Comment (savannahostrowski) [inline src/components/ListView.js]: We can assign `const {reports} = this.state;` at the top of of the render method before the return and then simplify this to just `reports`

File: src/components/ListCard.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+/**
+ * Get a string representing a timestamp in the format that Cloud Firestore sends us.
+ * Should be something we can call toString on, falsy, or an object with _longitude and
+ * _latitude as fields.
+ * TODO: once we handle neighborhoods properly, we should return a string representing
+ * the neighborhood here.
+ */
+const locationToString = (location) => {
```
Comment (savannahostrowski) [inline src/components/ListCard.js]: We can remove the parentheses around single params.

File: src/components/ListCard.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    display: 'flex',
+    padding: 0
+  }
+});
+
+/**
+ * Get a string for a timestamp in the format that Cloud Firestore sends us.
+ * Time should be an object with _nanoseconds and _seconds as fields.
+ */
+const timeToString = (time) => {
```
Comment (savannahostrowski) [inline src/components/ListCard.js]: Same here with the single param

File: src/components/ListView.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    flexDirection: 'column'
+  }
+});
+
+class ListView extends React.Component {
+
+  state = {reports: []};
+
+  componentDidMount() {
+    fetch(process.env.REACT_APP_GET_REPORTS, {method: 'GET', headers: {'Content-Length': 0}})
```
Comment (savannahostrowski) [inline src/components/ListView.js]: Just need to refactor with axios

File: src/components/ListView.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+  render() {
+    const {reports} = this.state;
+    const {classes} = this.props;
+    return (
+    <div className={classes.container}>
+    {
+      reports
+      ? reports.map((report) => <ListCard data={report.data} key={report.id}/>)
+      : 'Loading...'
```
Comment (savannahostrowski) [inline src/components/ListView.js]: You can actually throw some kind of spinner/progress bar component here.

https://material-ui.com/demos/progress/

**Comment by nachtm** [comment]
This is close @savannahostrowski if you want to merge in now so you can get your thing going -- otherwise, I'm just trying to clean up a little css so the background color is consistent on the loader. 
This is what it looks like now:
image

**Comment by savannahostrowski** [comment]
@nachtm Ah yes - here's [an example](https://github.com/spatialdev/osmquality-web/blob/138eece0b62afada69efd73d2f89ce9daba0c1de/src/components/CityProfileCard.js#L122) of what I've previously done for MQM UI stuff.  You can throw the circular spinner in a conditional and return it if the data isn't loaded. As soon as the data is loaded and you update `this.state`, the component will re-render with the list. This method also renders the spinner on the center of the page.

File: src/components/ListView.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (savannahostrowski) [inline src/components/ListView.js]: I'd order these like library package imports, material UI imports, and then custom component imports/CSS.

Comment (savannahostrowski) [inline src/components/ListView.js]: You can also import `{ Component } ` here so then in the class signature you can just extend `Component` without the dot.

File: src/components/ListView.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+});
+
+class ListView extends React.Component {
+
+  state = {reports: null};
+
+  componentDidMount() {
+    axios.get(process.env.REACT_APP_GET_REPORTS)
+      .then(reports => {
+        this.setState({...this.state, reports: reports.data});
```
Comment (savannahostrowski) [inline src/components/ListView.js]: You don't need `...this.state` here.

File: src/components/ListView.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      })
+      .catch(error => error);
+  }
+
+  render() {
+    const {reports} = this.state;
+    const {classes} = this.props;
+    return (
+    <div className={classes.container}>
+    {
```
Comment (savannahostrowski) [inline src/components/ListView.js]: I'd recommend just pulling this out to the top above the return statement as :

` if (!reports) {
return <CircularProgress/>;
}` 

then you can kill the ternary operator.

I think it's fine that the spinner is at the top.

File: src/components/Main.js [hunk 11]
```diff
@@ -2,14 +2,24 @@ import React from 'react';
 import { Switch, Route } from 'react-router-dom';
```
Comment (savannahostrowski) [inline src/components/Main.js]: Let's revert back the style changes on this file.

File: src/components/ListView.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import ListCard from '../components/ListCard';
+import { withStyles } from '@material-ui/core/styles';
+import axios from 'axios';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const styles = theme => ({
+  container: {
+    backgroundColor: 'grey',
```
Comment (savannahostrowski) [inline src/components/ListView.js]: Maybe light grey instead?

File: src/components/ListView.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    justifyContent: 'center'
+  }
+});
+
+class ListView extends React.Component {
+
+  state = {reports: null};
+
+  componentDidMount() {
+    axios.get(process.env.REACT_APP_GET_REPORTS)
```
Comment (savannahostrowski) [inline src/components/ListView.js]: Let's also remove this from the `.env` file and then just define a const outside the class where the cloud function lives on Firebase

File: src/components/ListCard.js [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+import CardMedia from '@material-ui/core/CardMedia';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+
+// TODO find something better
+const DEFAULT_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/thumb/7/79/2010-brown-bear.jpg/200px-2010-brown-bear.jpg";
+
+const styles = theme => ({
+  picture: {
+    width: 150,
```
Comment (savannahostrowski) [inline src/components/ListCard.js]: Do we need both min and regular height and widths here?

Comment (nachtm) [reply src/components/ListCard.js]: Nope!

**Pull request merged**